### PR TITLE
SONARJAVA-1046 Rule: S2384 Mutable members should not be stored or returned directly

### PR DESCRIPTION
**Pull request by mpaladin**

### CODE REVIEW

**Comment by m-g-sonar** [comment]
In order to make test cases more clear, I would remove what some useless part. Except for that LGTM!

File: java-checks/src/main/java/org/sonar/java/checks/MutableMembersUsageCheck.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+@Rule(
+  key = "S2384",
+  name = "Mutable members should not be stored or returned directly",
+  tags = {"cert", "cwe", "security", "unpredictable"},
+  priority = Priority.CRITICAL)
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.DATA_RELIABILITY)
+@SqaleConstantRemediation("5min")
+public class MutableMembersUsageCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+  private List<String> MUTABLE_TYPES = ImmutableList.of(
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/MutableMembersUsageCheck.java]: New MAJOR Rename this field name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00116)

File: java-checks/src/main/java/org/sonar/java/checks/MutableMembersUsageCheck.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    "java.util.Collection",
+    "java.util.Date",
+    "java.util.Hashtable"
+  );
+  private List<String> IMMUTABLE_TYPES = ImmutableList.of(
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/MutableMembersUsageCheck.java]: New MAJOR Rename this field name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00116)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 new issues:
* 2 major

**Pull request merged**